index.js

```diff
@@ -92,9 +95,7 @@ function TChannel(options) {
         self.logger.warn('server socket close');
     });
 
-    if (self.listening) {
```

Comment:
why did we remove this flag ? I'm using it in autobahn. Seperating out allocating and effects is nice.